src/keybind/key_names.cpp

```diff
@@ -0,0 +1,206 @@
+    {Key::key_0,            {"0"}},
+    {Key::key_1,            {"1"}},
+    {Key::key_2,            {"2"}},
+    {Key::key_3,            {"3"}},
+    {Key::key_4,            {"4"}},
+    {Key::key_5,            {"5"}},
+    {Key::key_6,            {"6"}},
+    {Key::key_7,            {"7"}},
+    {Key::key_8,            {"8"}},
+    {Key::key_9,            {"9"}},
+
+    {Key::key_a,            {"a", "A"}},
+    {Key::key_b,            {"b", "B"}},
+    {Key::key_c,            {"c", "C"}},
+    {Key::key_d,            {"d", "D"}},
+    {Key::key_e,            {"e", "E"}},
+    {Key::key_f,            {"f", "F"}},
+    {Key::key_g,            {"g", "G"}},
+    {Key::key_h,            {"h", "H"}},
+    {Key::key_i,            {"i", "I"}},
+    {Key::key_j,            {"j", "J"}},
+    {Key::key_k,            {"k", "K"}},
+    {Key::key_l,            {"l", "L"}},
+    {Key::key_m,            {"m", "M"}},
+    {Key::key_n,            {"n", "N"}},
+    {Key::key_o,            {"o", "O"}},
+    {Key::key_p,            {"p", "P"}},
+    {Key::key_q,            {"q", "Q"}},
+    {Key::key_r,            {"r", "R"}},
+    {Key::key_s,            {"s", "S"}},
+    {Key::key_t,            {"t", "T"}},
+    {Key::key_u,            {"u", "U"}},
+    {Key::key_v,            {"v", "V"}},
+    {Key::key_w,            {"w", "W"}},
+    {Key::key_x,            {"x", "X"}},
+    {Key::key_y,            {"y", "Y"}},
+    {Key::key_z,            {"z", "Z"}},
+
+    {Key::f1,               {"F1"}},
+    {Key::f2,               {"F2"}},
+    {Key::f3,               {"F3"}},
+    {Key::f4,               {"F4"}},
+    {Key::f5,               {"F5"}},
+    {Key::f6,               {"F6"}},
+    {Key::f7,               {"F7"}},
+    {Key::f8,               {"F8"}},
+    {Key::f9,               {"F9"}},
+    {Key::f10,              {"F10"}},
+    {Key::f11,              {"F11"}},
+    {Key::f12,              {"F12"}},
+    {Key::f13,              {"F13"}},
+    {Key::f14,              {"F14"}},
+    {Key::f15,              {"F15"}},
+    {Key::f16,              {"F16"}},
+    {Key::f17,              {"F17"}},
+    {Key::f18,              {"F18"}},
+    {Key::f19,              {"F19"}},
+    {Key::f20,              {"F20"}},
+    {Key::f21,              {"F21"}},
+    {Key::f22,              {"F22"}},
+    {Key::f23,              {"F23"}},
+    {Key::f24,              {"F24"}},
+
+    {Key::insert,           {"Insert"}},
+    {Key::clear,            {"Clear"}},
+
+    {Key::home,             {"Home"}},
+    {Key::end,              {"End"}},
+    {Key::pageup,           {"Page Up"}},
+    {Key::pagedown,         {"Page Down"}},
+
+    {Key::application,      {"Application"}},
+    {Key::printscreen,      {"Print Screen"}},
+
+    {Key::space,            {"Space"}},
+    {Key::tab,              {"Tab"}},
+    {Key::backspace,        {"Backspace"}},
+    {Key::delete_,          {"Delete"}},
+
+    {Key::equal,            {"=", "+"}},
```

Comment:
I don't read whole of this pull request, but do these keys work well for JIS keyboard, which is the most commonly used keyboard layout in Japan? E.g., `=` is next to `0`, and needs Shift key. `+` is next to `L` key and needs Shift key. JIS layout is far different from US layout, especially for symbol keys like this.

Reply:
It's meant to be completely compatible with the previous input system, only with an extra layer for adding new actions with keybindings. If one of the defaults provided is inconvenient on a particular key layout, it should be able to be rebound easily.

Reply:
Also, this PR isn't for replacing the old system, it is for adding the base classes before refactoring it.